webapp/src/pages/app/settings/server-status/index.tsx

```diff
@@ -0,0 +1,112 @@
+
+const ServerStatusPage = () => {
+  const { data: currentStatus, error, isLoading } = useGetServerStatusQuery();
+
+  const [selectedStatus, setSelectedStatus] = useState<ServerStatusType | undefined>(undefined);
+  const [selectedScope, setSelectedScope] = useState<ServerStatusScope>('ALL');
+
+  const getStatusColorScheme = (status: ServerStatusType) => {
+    switch (status) {
+      case 'WRITABLE':
+        return 'green';
+      case 'REPLICATION_ONLY':
+        return 'yellow';
+      case 'READ_ONLY':
+        return 'red';
+    }
+  };
+
+  const getStatusBadge = (status: ServerStatusType | undefined) => {
+    if (!status) {
+      return <Text ml="4">Not available</Text>;
+    }
+
+    return (
+      <Badge ml="4" px="3" py="1" fontSize="md" colorScheme={getStatusColorScheme(status)} borderRadius="md">
+        {status}
+      </Badge>
+    );
+  };
+
+  return (
+    <SettingView currentTab="Server Status">
+      <Deferred isLoading={isLoading} error={error}>
+        {() => (
+          <Box p="4">
+            <Flex mb="6" alignItems="center">
+              <Text fontSize="lg" fontWeight="bold">
+                Current Server Status:
+              </Text>
+              {getStatusBadge(currentStatus)}
+              <Spacer />
+            </Flex>
+
+            <Box mb="6">
+              <Text fontSize="lg" fontWeight="bold" mb="3">
+                Select New Server Status:
+              </Text>
+              <RadioGroup onChange={(val: ServerStatusType) => setSelectedStatus(val)} value={selectedStatus}>
+                <Stack direction={{ base: 'column', md: 'row' }} spacing="4">
+                  <Radio value="WRITABLE">Writable (Writable & Replicating)</Radio>
+                  <Radio value="REPLICATION_ONLY">Replication Only (Not Writable & Replicating)</Radio>
+                  <Radio value="READ_ONLY">Read Only (Not Writable & Not Replicating)</Radio>
+                </Stack>
+              </RadioGroup>
+            </Box>
+
+            <Box mb="6">
+              <Text fontSize="lg" fontWeight="bold" mb="3">
+                Apply Scope:
+              </Text>
+              <Select
+                value={selectedScope}
+                onChange={(e) => setSelectedScope(e.target.value as ServerStatusScope)}
+                width={{ base: '100%', md: '300px' }}
+              >
+                <option value="ALL">ALL (Propagate to all cluster servers)</option>
+                <option value="LOCAL">LOCAL (Apply only to this server)</option>
+              </Select>
+              <Text fontSize="sm" color="gray.500" mt="1">
+                Note: Setting &#39;Replication Only&#39; or &#39;Writable&#39; with{' '}
+                <Code variant="outline">ALL</Code> scope is not allowed if the cluster is currently not
+                replicating. Use <Code variant="outline">LOCAL</Code> scope for individual instance changes.
```

Comment:
Optional) Since we know the current state and the next state, we could optionally disable the button if a user tries to change the state from READ-ONLY a different state with ALL.
Having said this, as this is a page only for admins, I guess the current as-is is fine as well.